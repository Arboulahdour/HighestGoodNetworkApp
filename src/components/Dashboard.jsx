import React, { Component } from 'react'
import { connect } from 'react-redux'
import { getCurrentUser, getAllProjects, postTimeEntry } from '../actions'
import { getjwt } from '../services/loginService'
import { Card, Row, CardTitle, CardText, Col } from 'reactstrap'
import MonthlyEffort from './MonthlyEffort'
import Leaderboard from './Leaderboard'
import '../App.css'

class Dashboard extends Component {
  componentDidMount() {
    this.props.getCurrentUser(getjwt())
  }

  render() {
    return (
      <React.Fragment>
        <div>
          <Row>
            <Col sm={{ offset: 1, size: 7 }}></Col>
            <Col sm={{ size: 3 }}>
              <Card body inverse color='info'>
                <CardTitle>
                  <MonthlyEffort />
                </CardTitle>
                <CardText>
                  <div />
                </CardText>
              </Card>
            </Col>
          </Row>
          <Row style={{ marginTop: '20px' }}>
            <Col sm={{ offset: 1, size: 7 }}>
              <Card body inverse color='warning'>
                <CardTitle>Badges</CardTitle>
              </Card>
            </Col>
          </Row>
        </div>
      </React.Fragment>
    )
  }
}

const mapStateToProps = state => {
  return { state }
}

export default connect(mapStateToProps, { getCurrentUser })(Dashboard)
