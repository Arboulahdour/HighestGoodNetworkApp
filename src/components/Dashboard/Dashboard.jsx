import React from 'react';
import { Alert, Row, Col, Container } from 'reactstrap';
import Leaderboard from '../LeaderBoard';
import WeeklySummaryModal from '../WeeklySummary/WeeklySummaryModal';
import Badge from '../Badge';
import TeamMemberTasks from '../TeamMemberTasks/TeamMemberTasks'
import Timelog from '../Timelog/Timelog';
import '../../App.css';
import { connect } from 'react-redux';

const Dashboard = props => (
  <Container fluid>
    <Row>
      <Col sm={{ size: 12 }}>
        <Alert color="info">
          <b>Reminder</b>: Make sure to purge the cache or "hard" refresh the page in your browser
          if you don's see the changes you had merged with the "development" branch. This message
          will be removed before the site goes "live".
        </Alert>
      </Col>
    </Row>
    <Row>
      <Col lg={{ size: 7 }}>&nbsp;</Col>
      <Col lg={{ size: 5 }}>
        <WeeklySummaryModal />
      </Col>
    </Row>
    <Row>
      <Col lg={{ size: 5 }} className="order-sm-12">
        <Leaderboard />
      </Col>
      <Col lg={{ size: 7 }} className="left-col-dashboard order-sm-1">
        <div className="my-2">
          <Timelog/>
        </div>
        <div className="my-2">
          <TeamMemberTasks />
        </div>
        <Badge userId={props.auth.user.userid} />
      </Col>

    </Row>
  </Container >
);

const mapStateToProps = state => ({
  auth: state.auth,
})

export default connect(mapStateToProps)(Dashboard);