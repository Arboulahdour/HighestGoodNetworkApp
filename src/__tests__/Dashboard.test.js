import { shallow } from 'enzyme';
import React from 'react';
import Dashboard from '../components/Dashboard';


describe('Dashboard tests', () => {
  let dashBoardMountedPage;
  beforeEach(() => {
    dashBoardMountedPage = shallow(<Dashboard />);
  });

  xit('should render a leaderboard', () => {
    const leaderboard = dashBoardMountedPage.find('Leaderboard').length;
    expect(leaderboard).toBe(1);
  });
});
