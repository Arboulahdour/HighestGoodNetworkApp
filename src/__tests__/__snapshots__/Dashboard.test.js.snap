// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard  should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-fluid"
  >
    <div
      class="row"
    >
      <div
        class="col-sm-12"
      >
        <div
          class="alert alert-info fade"
          role="alert"
        >
          <b>
            Reminder
          </b>
          : Make sure to purge the cache or "hard" refresh the page in your browser if you don's see the changes you had merged with the "development" branch. This message will be removed before the site goes "live".
        </div>
      </div>
      <div
        class="col-lg-9"
      >
        <div
          class="container-fluid"
        >
          <div
            class="text-center row"
            data-testid="loading"
          >
            <div
              class="container-fluid"
            >
              <div
                class="fa-5x"
              >
                <i
                  class="fa fa-spinner fa-pulse"
                />
              </div>
            </div>
          </div>
        </div>
        <table
          class="table"
        >
          <thead>
            <tr>
              <th>
                Status
              </th>
              <th>
                Name
              </th>
              <th>
                intangibletime
              </th>
              <th>
                Progress
              </th>
              <th>
                totaltime
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <div
                  style="background-color: orange; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5edf141c78f1380017b829a6"
                >
                  Dev Admin
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: orange;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                6
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5bef421d908c1b0038d06dbd"
                >
                  Adam Capdeville
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5eecff1e80d46831c828ed86"
                >
                  Adam Volunteer
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5eddbcd353a0480017164a57"
                >
                  Admin New
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5e9fa3d9b5dc3c2e48c10989"
                >
                  Alvaro Hern√°ndez
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5e9fa2fbb3acfb7f9a4d1d8c"
                >
                  Alvaro user1
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5ec95665c77af83f146f339b"
                >
                  Andon Testing
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5ee257d03f0d66001704f8aa"
                >
                  Dev Volunteer
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5e2cc7087a91aaf820510a56"
                >
                  Dev user1
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5ab1713bba0e82002b182206"
                >
                  Jae Sabol
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5f20d19160bd1734f069dadb"
                >
                  Jerry Administrator
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5f20d18660bd1734f069dada"
                >
                  Jerry Manager
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5f20d17260bd1734f069dad9"
                >
                  Jerry Volunteer
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5f1fb5d9974f146edcfa741b"
                >
                  Jerry Zhang
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5f31dcb9a1a909eadee0eecb"
                >
                  Jun Hao
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5ea5ba581446e1164098fa7b"
                >
                  New Manager
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5eb2c7e9f0eb3233382d232e"
                >
                  NewAdmin User
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5ede6be453a0480017164a5b"
                >
                  Nithesh Admin
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5edda90b53a0480017164a4e"
                >
                  Nithesh TEK
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5f150b7599b77700174e69fb"
                >
                  Nithesh Train
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5e28bdaab9ec7807f278aea3"
                >
                  Rafel Sabol
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5e2cc6df7a91aaf820510a55"
                >
                  Sidd Admin
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5e2e3d3d51ddad1693d79279"
                >
                  Simon Xiong
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5e115150c472127fee9ed87d"
                >
                  Tengxiao Wang
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5c4cc2109487b0003924f1e3"
                >
                  Test Admin
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5e2a76cb0441c30039a6909a"
                >
                  Test Cat
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5c63038e2b88c00039ed79e6"
                >
                  Test User
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5ea5b0d81446e1164098fa7a"
                >
                  Test User99
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5e66cdf5336b3af7f2466d2b"
                >
                  Wen Zhang
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                <div
                  style="background-color: red; width: 15px; height: 15px; border-radius: 7.5px;"
                />
              </td>
              <th
                scope="row"
              >
                <a
                  href="/userprofile/5bde804fb2d9e00016c47fd1"
                >
                  sddf fsdfsdf
                </a>
              </th>
              <td>
                0
              </td>
              <td>
                <div
                  class="progress"
                  style="background-color: red;"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="10"
                    class="progress-bar"
                    role="progressbar"
                    style="width: 10%;"
                  />
                </div>
              </td>
              <td>
                0
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="col-lg-3"
      >
        <div
          class="card text-white card-body bg-info"
        >
          <div
            class="card-body text-white"
          >
            <h5
              class="card-title"
            >
              Monthly Efforts
            </h5>
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
